src/osvr/Server/ConfigureServerFromFile.cpp

```diff
@@ -0,0 +1,175 @@
+                for (auto const &plugin : srvConfig.getSuccessfulPlugins()) {
+                    log->info() << " - " << plugin;
+                }
+            }
+            if (!srvConfig.getFailedPlugins().empty()) {
+                log->warn() << "Failed to load the following plugins:";
+                for (auto const &pluginError : srvConfig.getFailedPlugins()) {
+                    log->warn() << " - " << pluginError.first << "\t"
+                        << pluginError.second;
+                }
+            }
+        }
+
+        {
+            log->info() << "Instantiating configured drivers...";
+            bool success = srvConfig.instantiateDrivers();
+            if (!srvConfig.getSuccessfulInstantiations().empty()) {
+                log->info() << "Successes:";
+                for (auto const &driver :
+                    srvConfig.getSuccessfulInstantiations()) {
+                    log->info() << " - " << driver;
+                }
+            }
+            if (!srvConfig.getFailedInstantiations().empty()) {
+                log->error() << "Errors:";
+                for (auto const &error : srvConfig.getFailedInstantiations()) {
+                    log->error() << " - " << error.first << "\t"
+                        << error.second;
+                }
+            }
+        }
+
+        if (srvConfig.processExternalDevices()) {
+            log->info()
+                << "External devices found and parsed from config file.";
+        }
+
+        if (srvConfig.processRoutes()) {
+            log->info() << "Routes found and parsed from config file.";
+        }
+
+        if (srvConfig.processAliases()) {
+            log->info() << "Aliases found and parsed from config file.";
+        }
+
+        if (srvConfig.processDisplay()) {
+            log->info()
+                << "Display descriptor found and parsed from config file.";
+        } else {
+            log->info()
+                << "Using OSVR HDK for display configuration. "
+                "Did not find an alternate valid 'display' object in config "
+                "file.";
+        }
+
+        if (srvConfig.processRenderManagerParameters()) {
+            log->info() << "RenderManager config found and parsed from the "
+                "config file.";
+        }
+
+        log->info() << "Triggering automatic hardware detection...";
+        ret->triggerHardwareDetect();
+
+        return ret;
+    }
+
+    ServerPtr configureServerFromFile(std::string const &configName) {
+        auto log =
+            ::osvr::util::log::make_logger(::osvr::util::log::OSVR_SERVER_LOG);
+
+        ServerPtr ret;
+        log->info() << "Attempting to load config file '" << configName << "'.";
+        std::ifstream config(configName);
+        if (!config.good()) {
+            log->error() << "Config file '" << configName << "' not found";
+            return nullptr;
+        }
+
+        std::stringstream sstr;
```

Comment:
@rpavlik Yeah, this just loads the whole file into a string. It's the best way I could find to do it.